File: backend/app/models/conflict.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel, Field


class ConflictVote(BaseModel):
    user_id: str = Field(alias="userId")
    at: Optional[datetime] = None

    class Config:
        populate_by_name = True


class ConflictOption(BaseModel):
    key: str  # 'a', 'b', etc.
    title: str
    description: Optional[str] = None
    votes: List[ConflictVote] = Field(default_factory=list)


class ConflictBase(BaseModel):
    options: List[ConflictOption]


class ConflictCreate(ConflictBase):
    message_id: str


class ConflictInDB(ConflictBase):
    id: Optional[str] = Field(default=None, alias="_id")
    trip_id: Optional[str] = Field(default=None, alias="tripId")
    message_id: Optional[str] = Field(default=None, alias="messageId")
    created_at: Optional[datetime] = None

    class Config:
        populate_by_name = True


class Conflict(ConflictBase):
    id: str
    trip_id: str
    message_id: str
    created_at: datetime

    class Config:
        from_attributes = True

File: backend/src/models/conflict.ts
import { z } from 'zod';

export const ConflictVoteSchema = z.object({
  userId: z.string(),
  at: z.date().optional(),
});

export const ConflictOptionSchema = z.object({
  key: z.string(),
  title: z.string(),
  description: z.string().optional(),
  votes: z.array(ConflictVoteSchema).default([]),
});

export const ConflictBaseSchema = z.object({
  options: z.array(ConflictOptionSchema),
});

export const ConflictCreateSchema = ConflictBaseSchema.extend({
  message_id: z.string(),
});

export type ConflictVote = z.infer<typeof ConflictVoteSchema>;
export type ConflictOption = z.infer<typeof ConflictOptionSchema>;
export type ConflictBase = z.infer<typeof ConflictBaseSchema>;
export type ConflictCreate = z.infer<typeof ConflictCreateSchema>;

export interface ConflictInDB extends ConflictBase {
  _id?: string;
  tripId?: string;
  messageId?: string;
  createdAt?: Date;
}

export interface Conflict extends ConflictBase {
  id: string;
  trip_id: string;
  message_id: string;
  created_at: Date;
}
